.github/workflows/build-ubi-dependency.yml

```diff
@@ -0,0 +1,137 @@
+name: Build UBI ppc64le Dependency
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - build/dependencies/Dockerfile.ubi-ppc64le
+  workflow_dispatch:
+    inputs:
+      nginx_version:
+        type: string
+        description: "NGINX Version to build for"
+        required: false
+      force:
+        type: boolean
+        description: "Force rebuild"
+        required: false
+        default: false
+
+env:
+  IMAGE_NAME: ghcr.io/nginxinc/dependencies/nginx-ubi-ppc64le
+
+concurrency:
+  group: ${{ github.ref_name }}-ubi-ppc64le-build
+  cancel-in-progress: true
+
+jobs:
+  checks:
+    name: Check versions
+    runs-on: ubuntu-22.04
+    permissions:
+      packages: read
+      contents: read
+    strategy:
+      fail-fast: false
+    outputs:
+      nginx_version: ${{ steps.var.outputs.nginx_version }}
+      njs_version: ${{ steps.var.outputs.njs_version }}
+      target_exists: ${{ steps.var.outputs.target_image_exists }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@0d4c9c5ea7693da7b068278f7b52bda2a190a446 # v3.2.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Output Variables
+        id: var
+        run: |
+          if [ -n "${{ inputs.nginx_version }}" ]; then
+            nginx_v=${{ inputs.nginx_version }}
+          else
+            nginx_v=$(grep -m1 'FROM nginx:' <build/dependencies/Dockerfile.ubi-ppc64le | cut -d '@' -f1 | awk -F'[: ]' '{print $3}')
+          fi
```

Comment:
Should we add an `echo` here to if the first condition doesn't happen?
Some thing like "nginx version not found, getting version dockerfile"?

Reply:
The `[ -n "${{ inputs.nginx_version }}" ]` will only evaluate to true when the workflow is run manually, so the `else` is the default. It should also be clear which is used from the echo on line 70.